pkg/handler/handler.go

```diff
@@ -390,8 +403,16 @@ func (m *Modifier) MutatePod(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResp
 		}
 	}
 
-	klog.V(3).Infof("Pod was mutated. %s",
-		logContext(pod.Name, pod.GenerateName, pod.Spec.ServiceAccountName, pod.Namespace))
+	// TODO: klog structured logging can make this better
+	if changed {
+		klog.V(3).Infof("Pod was mutated. %s",
+			logContext(pod.Name, pod.GenerateName, pod.Spec.ServiceAccountName, pod.Namespace))
+	} else {
+		klog.V(3).Infof("Pod was not mutated. Reason: %s. %s",
+			"Required volume mounts and env variables were already present",
```

Comment:
By reading the code, it looks like the reason is most likely the env variables are already present (missing volume alone won't mutate the pod).

Reply:
I did not fully understand this. If volume is missing on the pod we do mutate the pod here - https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/pkg/handler/handler.go#L308. There are similar lines for adding volume mounts on eligible containers here https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/pkg/handler/handler.go#L242

Reply:
Nit: the reason is hard-coded, and yet using a string substitution "%s"...  It should just be one substitution for the logContext?

Reply:
Ahh yes. Small change. Will fix.

Reply:
My bad, was trying to point out missing "volume mounts" in the containers isn't enough to mutate since it will return early if env variables are set. It does get mutated if the "volumes" are missing.

https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/pkg/handler/handler.go#L242